src/production/module/SplitEntries.ts

```diff
@@ -0,0 +1,50 @@
+import { Module } from '../../core/Module';
+import { IProductionContext } from '../ProductionContext';
+import { IImport } from './ImportReference';
+import { ModuleType } from './ModuleTree';
+
+export interface ISplitEntry {
+  entry: Module;
+  modules: Array<Module>;
+  references: Array<IImport>;
+}
+
+export interface ISplitEntryProps {
+  module: Module;
+  productionContext: IProductionContext;
+  subModules: Array<Module>;
+}
+
+export function SplitEntry(props: ISplitEntryProps): ISplitEntry {
+  const { module, subModules } = props;
+  module.moduleTree.moduleType = ModuleType.SPLIT_MODULE;
+
+  return {
+    entry: module,
+    modules: subModules,
+    references: module.moduleTree.dependants,
+  };
+}
+
+export interface ISplitEntries {
+  entries: Array<ISplitEntry>;
+  ids: { [key: number]: boolean };

```

Comment:
not sure if this is a good name for an object of moduleIds... It's used for quick lookup if a module is a splitEntry

Reply:
ids is fine. Just use `Record<number, boolean>` looks better ;-)

Reply:
✅ done